templates/commands/render_action.go

```diff
@@ -37,65 +37,85 @@ import (
+		walkFrom := filepath.Join(scratchDir, relPath)
+		if _, err := rfs.Stat(walkFrom); err != nil {
+			if os.IsNotExist(err) {
+				return model.ErrWithPos(pos, `path %q doesn't exist in the scratch directory, did you forget to "include" it first?"`, relPath) //nolint:wrapcheck
+			}
+			return model.ErrWithPos(pos, "Stat(): %w", err) //nolint:wrapcheck
 		}
 
-		oldBuf, err := rfs.ReadFile(path)
-		if err != nil {
-			return model.ErrWithPos(pos, "Readfile(): %w", err) //nolint:wrapcheck
-		}
+		err = filepath.WalkDir(walkFrom, func(path string, d fs.DirEntry, err error) error {
+			if err != nil {
+				// There was some filesystem error. Give up.
+				return model.ErrWithPos(pos, "%w", err) //nolint:wrapcheck
+			}
+			if d.IsDir() {
+				return nil
+			}
 
-		relToScratchDir, err := filepath.Rel(scratchDir, path)
-		if err != nil {
-			return model.ErrWithPos(pos, "Rel(): %w", err) //nolint:wrapcheck
-		}
+			if _, ok := seen[path]; ok {
+				// File already processed.
+				return nil
+			}
+			oldBuf, err := rfs.ReadFile(path)
+			if err != nil {
+				return model.ErrWithPos(pos, "Readfile(): %w", err) //nolint:wrapcheck
+			}
 
-		// We must clone oldBuf to guarantee that the callee won't change the
-		// underlying bytes. We rely on an unmodified oldBuf below in the call
-		// to bytes.Equal.
-		newBuf, err := v(bytes.Clone(oldBuf))
-		if err != nil {
-			return fmt.Errorf("when processing template file %q: %w", relToScratchDir, err)
-		}
+			relToScratchDir, err := filepath.Rel(scratchDir, path)
+			if err != nil {
+				return model.ErrWithPos(pos, "Rel(): %w", err) //nolint:wrapcheck
+			}
 
-		if bytes.Equal(oldBuf, newBuf) {
-			// If file contents are unchanged, there's no need to write.
-			return nil
-		}
+			// We must clone oldBuf to guarantee that the callee won't change the
+			// underlying bytes. We rely on an unmodified oldBuf below in the call
+			// to bytes.Equal.
+			newBuf, err := v(bytes.Clone(oldBuf))
+			if err != nil {
+				return fmt.Errorf("when processing template file %q: %w", relToScratchDir, err)
+			}
 
-		// The permissions in the following WriteFile call will be ignored
-		// because the file already exists.
-		if err := rfs.WriteFile(path, newBuf, ownerRWXPerms); err != nil {
-			return model.ErrWithPos(pos, "Writefile(): %w", err) //nolint:wrapcheck
-		}
-		logger.Debugw("wrote modification", "path", path)
+			seen[path] = struct{}{}
 
-		return nil
-	})
+			if bytes.Equal(oldBuf, newBuf) {
+				// If file contents are unchanged, there's no need to write.
+				return nil
+			}
+
+			// The permissions in the following WriteFile call will be ignored
+			// because the file already exists.
+			if err := rfs.WriteFile(path, newBuf, ownerRWXPerms); err != nil {
+				return model.ErrWithPos(pos, "Writefile(): %w", err) //nolint:wrapcheck
+			}
+			logger.Debugw("wrote modification", "path", path)
+
+			return nil
+		})
+		if err != nil {
+			return fmt.Errorf("walkAndModify error for action: %w", err)
```

Comment:
Maybe just `return err` here? I don't see the extra text adding much context. Typically if we annotate an error with the name of the function that failed, we do it in the caller, like:

```
if err := doThing(); err != nil {
  return fmt.Errorf("doThing: %w", err)
}
```

Reply:
The linter threw a fit so I did that, should I just disable the linter for that line?

Reply:
Yeah, do the usual //nolint: thing.